### ISSUE TITLE
about the database

### ISSUE THREAD

**Issue by PoonamZ** [shreyaspadhy/UNet-Zoo#8]
@shreyaspadhy  I am using brats 2017 database which is in .nii format. It is necessary to convert that data into .png format?

**Comment by TheCarmelGuy** [comment]
The data.py module assumed that your data and ground truth segmentation maps are already sliced (in png format) into 2D images. '.nii' files are 3D volumes so yes, I believe you would need to convert them to .png

**Comment by shreyaspadhy** [comment]
Hi, for our particular implementation, we'd manually converted the .nii files into .png files where each .png corresponded to one z-slice of a patient. If you want to directly use .nii files, you could look into NiBabel (http://nipy.org/nibabel/gettingstarted.html), and modify the Dataset function in 'data.py' to suit your personal way of handling the dataset. Let me know if you need any more clarifications!

**Comment by shreyaspadhy** [comment]
As a follow up, to use the code here for any form of dataset, you simply have to modify the __init__(), __getitem__() and __len__() functions within the BratsDatasetUNet() Class.

In general, to define a Dataset Class for Pytorch, you want __getitem__(index) to return one (image, mask) pair from your entire dataset as FloatTensors/numpy arrays, and __len__() should return the total number of images in the dataset. Beyond this, you can use the remaining training code unchanged.
